### ISSUE TITLE
Should we name `data.formulas` to `data.derive`?

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#664]
Forked from #655: 
Recently, we introduced `data.formulas` property for deriving calculated field.   I propose that we use `data.derive` instead of `data.formulas` because the data spec would be like this. 

```js
data: {
   derive: [
     {field: ..., expr: …},
     {field: …, expr: …}
   }]
}
```
This can be easily interpret as derive a field with a particular expression, which is more intuitive than:

```js
data: {
   formulas: [
     {field: ..., expr: …},
     {field: …, expr: …}
   }]
}
```

Since this property is first introduced in the unreleased `0.8`, changing it would not affect many people.

**Comment by domoritz** [comment]
I'm against it. Formula is a vega term and we should aim for consistency. This way people can better google for help.

**Comment by kanitw** [comment]
> we should aim for consistency.

Consistency isn't always win, especially when they are different things.  
Here we are talking about property name.    `derive` is semantically more meaningful as this is for deriving new fields.
While for Vega, `formula` is a type of transform so it's sensible name.  

> This way people can better google for help.

Also, this is a simple feature so we should have solid docs for this so people do not need to "google for help".   Even if people want to google, people might want to google "How do I derive new field in Vega-lite". Thus, I don't buy the "better google for help" argument.

**Comment by kanitw** [comment]
@jheer  Do you have any opinion on this naming issue?

**Comment by kanitw** [comment]
We decide to do `calculate`

**Comment by kanitw** [comment]
Fixed in #747
